pandas/core/dtypes/missing.py

```diff
@@ -640,6 +640,9 @@ def is_valid_na_for_dtype(obj, dtype: DtypeObj) -> bool:
         # Numeric
         return obj is not NaT and not isinstance(obj, (np.datetime64, np.timedelta64))
 
+    elif dtype == np.dtype(str):
```

Comment:
hmm

Reply:
what is this about?

Reply:
do any tests rely on this?

Reply:
Yep, unfortunately

```
FAILED pandas/tests/frame/methods/test_astype.py::TestAstypeCategorical::test_astype_categorical_to_string_missing
```

It checks if ``df.astype(str)`` is equal to ``df.astype("category").astype(str)``

If we do not check this here, the float ``np.nan`` is considered as a correct missing value for numpy string dtypes instead of the object ``np.nan``, which is used when ``df.astype(str)`` is called immediately.

Reply:
got it, can you add a comment (like we do below), also prob should define `_dtype_str` (and `_dtype_object`) at the module level and use here (can be followup)

Reply:
Added the comment and moved the creation of the dtypes to the module level